web/src/components/shared/AirgapRegistrySettings.tsx

```diff
@@ -143,16 +187,20 @@ class AirgapRegistrySettings extends Component {
     }
   };
 
-  handleFormChange = (field, val) => {
-    let nextState = {};
+  handleFormChange = (field: string, val: string | boolean) => {
+    let nextState: { [key: string]: string | boolean } = {};
     nextState[field] = val;
 
     if (this.props.app?.isAirgap && field === "isReadOnly" && !val) {
       // Pushing images in airgap mode is not yet supported, so registry name cannot be changed.
-      nextState["hostname"] = this.state.originalRegistry.hostname;
-      nextState["namespace"] = this.state.originalRegistry.namespace;
+      nextState["hostname"] = this.state.originalRegistry!.hostname;
+      nextState["namespace"] = this.state.originalRegistry!.namespace;
     }
 
+    //TODO: understand this better
+    // Argument of type '{ [key: string]: string | boolean; }' is not assignable to parameter of type 'State | Pick<State, keyof State>
+    // | ((prevState: Readonly<State>, props: Readonly<Props>) => State | Pick<...> | null) | null'.
+    // @ts-ignore
     this.setState(nextState, () => {
```

Comment:
really couldn't figure this one out :(

Reply:
We can come back to it. I think can you use `Partial` or `Pick` but I'm not sure which is the right one. I [used](https://github.com/replicatedhq/kots/blob/82c8c5b06191762a1e447438a7ebf056895a1602/web/src/Root.tsx#L109) `Partial` in `Root.tsx` for now. 

https://www.typescriptlang.org/docs/handbook/utility-types.html#partialtype

Pick is probably better but I don't know how to use it yet.